content/blog/usemodels-0-0-1/index.Rmd

```diff
@@ -0,0 +1,106 @@
+ 
+ * [Create a recipe](https://www.tmwr.org/recipes.html) that, at minimum, centers and scales the predictors. For some data sets, we also need to create dummy variables from any factor-encoded predictor columns. 
+ 
+ * Define a [resampling scheme](https://www.tmwr.org/resampling.html) for our data. 
+ 
+ * Choose a function from the [tune package](https://tune.tidymodels.org/), such as `tune_grid()`, to optimize the parameters. For grid search, we'll also need a grid of candidate parameter values (or let the function choose one for us). 
+ 
+We recognize that this might be more code than you would have had to write compared to a package like `caret`. However, the tidymodels ecosystem enables a wider variety of modeling techniques and has slightly higher complexity.
+
+usemodels automates much of this code infrastructure. For example: 
+
+```{r glmnet-basic, results = 'hide', prompt = TRUE}
+library(usemodels)
+use_glmnet(mpg ~ ., data = mtcars)
+```
+ 
+which produces the terminal output:
+
+```{r glmnet-basic-code, echo = FALSE, comment=""}
+library(usemodels)
+use_glmnet(mpg ~ ., data = mtcars)
+```
+ 
+This can be copied to the source window and edited. Some notes: 
+
+ * For this model, it is possible to prescribe a default grid of candidate tuning parameter values that  work well about 90% of the time. For other models, the grid might be data-driven. In these cases, our functions `tune` functions can estimate an appropriate grid. 
+ 
+ * The extra recipes steps are the [recommend preprocessing](https://www.tmwr.org/pre-proc-table.html) for this model. Since this varies from model-to-model, the recipe template will contain the minimal required steps. Your data might require additional operations. 
+ 
+ * One thing that _should not be automated_ is the choice of resampling method. The code templates require the user to choose the `rsample` function that is appropriate. 
+ 
+In case you are unfamiliar with the model and its preprocessing needs, a `verbose` option prints comments that explain _why_ some steps are included. For the glmnet model, the comments added to the recipe state: 
+
+> Regularization methods sum up functions of the model slope coefficients. Because of this, the predictor variables should be on the same scale. Before centering and scaling the numeric predictors, any predictors with a single unique value are filtered out. 
+
+Let's look at another example. The `ad_data` data set in the modeldata package has rows for 333 patients with a factor outcome for their level of cognitive impairment (e.g., Alzheimer's disease). There are also a categorical predictor in the data, the Apolipoprotein E genotype, which has six levels. Let's suppose the `Genotype` column was encoded as character (instead of being a factor). This might be a problem if the resampling method samples out a level from the data used to fit the model.
```

Comment:
```suggestion
Let's look at another example. The `ad_data` data set in the modeldata package has rows for 333 patients with a factor outcome for their level of cognitive impairment (e.g., Alzheimer's disease). There is also a categorical predictor in the data, the Apolipoprotein E genotype, which has six levels. Let's suppose the `Genotype` column was encoded as character (instead of being a factor). This might be a problem if the resampling method samples out a level from the data used to fit the model.
```